src/client/gui/TextBox.cpp

```diff
@@ -0,0 +1,116 @@
+#include "TextBox.hpp"
+#include "resource/FontManager.hpp"
+namespace bm::gui {
+TextBox::TextBox(const sf::Vector2f &position, const sf::Vector2f &size,
+                 const std::string &placeholder, unsigned int characterSize,
+                 std::size_t maxLength)
+    : m_isActive(false),
+      m_text{resource::FontManager::get().at(resource::FontId::MENU)},
+      m_maxLength(maxLength),
+      m_placeHolder{resource::FontManager::get().at(resource::FontId::MENU)},
+      m_cursorActive(true), m_cursorPosition(0) {
+
+  m_boxShape.setPosition(position);
+  m_boxShape.setSize(size);
+  m_boxShape.setFillColor(sf::Color::White);
+  m_boxShape.setOutlineThickness(3.f);
+  m_boxShape.setOutlineColor(sf::Color::Black);
+
+  auto &fontManager = resource::FontManager::get();
+
+  m_text.setString("");
+  m_text.setCharacterSize(characterSize);
+  m_text.setFillColor(sf::Color::Black);
+  auto textRect = m_text.getLocalBounds();
+  m_text.setOrigin(sf::Vector2f(textRect.position.x + textRect.size.x / 2.f,
+                                textRect.position.y + textRect.size.y / 2.f));
+  m_text.setPosition(
+      sf::Vector2f(m_boxShape.getPosition().x, m_boxShape.getPosition().y));
```

Comment:
setPosition accepts sf::Vector2f, why don't you just do:
m_text.setPosition(m_boxShape.getPosition())
instead?